src/extras/module-types.js

```diff
@@ -33,6 +33,9 @@
       if (cssContentType.test(contentType))
         return res.text()
         .then(function (source) {
+          source = source.replace(/url\(([ '"]?)([^:]*?)\1\)/g, function (match, quotes, id) {
```

Comment:
I'm not sure if CSS can be accurately parsed with regex like this. Here are a few edge cases I'm thinking about:

- `url(data:...)`
- `url(https://example.com/images/myImg.jpg)` (no quotes)
- `url("")` inside of a comment
- `url(#idOfSvgPath)`
- if it's somehow possible to inject a whitespace character somewhere that messes things up
- Character escaping within css

There might be more situations I'm unaware of, too. The regex here matches all of the above, although I think the bottom two would ideally not be matched?

The other disadvantage of this approach is the performance impact of regex string replacement on large css files.

Thoughts on these edge cases? Can you think of others?

Reply:
> url(data:...)

Yes.

> url(https://example.com/images/myImg.jpg) (no quotes)

Yes.

> url("") inside of a comment

No. Check if `url("")` is inside of a comment or not will make RegEx too complicated, but ignore empty `url("")` is necessary.

> url(#idOfSvgPath)

Yes.

> if it's somehow possible to inject a whitespace character somewhere that messes things up

Yes. According to the CSS specification, whitespace character must be escaped with a backslash.
![image](https://user-images.githubusercontent.com/2867188/116661762-88953d00-a9c7-11eb-8e81-dea2bc622311.png)


> Character escaping within css

Yes. But missing the edge case of `\)`. For example, partial url of `url(images/myImg\).jpg);` should be `images/myImg\).jpg` rather than `images/myImg\`.

**I have made a commit to fix these edge cases. Please take a look.**

Below are edge cases I have thought.
[https://regex101.com/r/uQkG3W/3/](https://regex101.com/r/uQkG3W/3/)
![image](https://user-images.githubusercontent.com/2867188/116662218-12dda100-a9c8-11eb-9815-4d0bd8af0588.png)

And there is a use case which solve the problem of relative url inside font-awesome.
[https://codesandbox.io/s/wizardly-cohen-t12yk](https://codesandbox.io/s/wizardly-cohen-t12yk)
![image](https://user-images.githubusercontent.com/2867188/116662719-b4fd8900-a9c8-11eb-8b3d-1a3c8f370507.png)